### PR TITLE
Note add and delete functionality

### PR DESCRIPTION
**Pull request by sushil-mod**
# Added Note add and delete functionality
how to check :-

- for Add Note : : go to login -> login as a gues -> then on mynotes create note and click on add button -> your note is created
- for Delete Note : : click on trash icon -> it will go to the Trash page -> navigate to trash page using sidebar-> then click restore -> it will restore the note -> click on delete icon it will permanently delete
### css and jsx
-  MyNotes 
-  CreateNoteCard. 
-  NoteCard 

## screenshot 
noteweb

### CODE REVIEW

File: src/components/card/NoteCard.jsx [hunk 1]
```diff
@@ -1,6 +1,85 @@
...
+export const NoteCard =( {noteItem }  )=> {
+
+  // const { _id , title ,description ,pinStatus ,color, tags, priority } = noteitem;
+  console.log(noteItem.title);
+  const { authState } = useAuthorizer();
+  const { setNoteList } = useNotes();
+  
+
+  const updateNoteItem = async(changedNote,id ) =>{
+    console.log( " changedNote " , changedNote ) ;
```
Comment (munjalSaksham71) [inline src/components/card/NoteCard.jsx:19]: Remove those logs

File: src/components/card/NoteCard.jsx [hunk 2]
```diff
@@ -1,6 +1,85 @@
...
+    try {
+      const response = await axios.post(`/api/notes/${id}`,{note:changedNote},{headers:{authorization:authState.encodedToken}});
+      console.log("update Note",response.data.notes);
+      setNoteList(response.data.notes);
+      
+    } catch (error) {
+      console.log("error");
```
Comment (munjalSaksham71) [inline src/components/card/NoteCard.jsx:26]: here you can alert the error

File: src/components/card/NoteCard.jsx [hunk 3]
```diff
@@ -1,6 +1,85 @@
...
+    }
+    
+  }
+
+  const deleteNote = async(id) => {
+    try {
+      const response = await axios.delete(`/api/notes/${id}`,{headers:{authorization:authState.encodedToken}});
+      console.log("delete",response);
```
Comment (munjalSaksham71) [inline src/components/card/NoteCard.jsx:34]: Remove this also

File: src/context/NotesContext.jsx [hunk 4]
```diff
@@ -1,27 +1,51 @@
...
+        trashStatus:false
     }
     
     const [ note ,setNote ] =useState(initialNote);
     const [ labelArray , setLabelArray ] =useState([]);
     const [bgColor,setBgColor] = useState("color-white");
-console.log(note);
+    const [noteList , setNoteList] = useState([]);
+
+    console.log(note);
```
Comment (munjalSaksham71) [inline src/context/NotesContext.jsx:26]: Remove this also

**Review by munjalSaksham71 — APPROVED** [review]
Overall it looks good, Just remove these unnecessary Logs and while catching error you can alert the error or set a new state and show the error in UI

**Comment by sushil-mod** [comment]
> Overall it looks good, Just remove these unnecessary Logs and while catching error you can alert the error or set a new state and show the error in UI

yeah will add toast feature for errors . also will reomve the console logs

File: src/components/card/NoteCard.jsx [hunk 5]
```diff
@@ -1,6 +1,85 @@
...
+  console.log(noteItem.title);
+  const { authState } = useAuthorizer();
+  const { setNoteList } = useNotes();
+  
+
+  const updateNoteItem = async(changedNote,id ) =>{
+    console.log( " changedNote " , changedNote ) ;
+    try {
+      const response = await axios.post(`/api/notes/${id}`,{note:changedNote},{headers:{authorization:authState.encodedToken}});
+      console.log("update Note",response.data.notes);
```
Comment (atharvabhanage02) [inline src/components/card/NoteCard.jsx:22]: you can remove the console.logs

**Review by atharvabhanage02 — APPROVED** [review]
- Code looks clean and well structured
- Liked the File structure
- UI looks clean
- Covered the edge cases well

**Pull request merged**